engine/consensus/dkg/reactor_engine.go

```diff
@@ -284,8 +284,13 @@ func (e *ReactorEngine) handleEpochCommittedPhaseStarted(currentEpochCounter uin
 
 		return
 	}
+	if currentState == flow.RandomBeaconKeyCommitted {
+		// this can happen if a healthy node which succeeded the DKG restarts in the EpochCommit phase
+		log.Debug().Msg("checking beacon key consistency after EpochCommit: observed committed beacon key for most recent dkg - exiting")
```

Comment:
Is this only for better traces? Functionality it doesn't change anything.

Reply:
Yes, this is for more accurate logs - there is no functional change.

Reply:
I think it does change something, because it shortcuts the sanity check that all (repeated) calls to committing the same epoch don't have inconsistent information. See [my comment below](https://github.com/onflow/flow-go/pull/6898#discussion_r1920683380) for details.